docs/performance-comparison/rosbag2-plugin-comparison.md

```diff
@@ -0,0 +1,152 @@
+- `sqlite_resilient`: The "resilient" preset profile of the SQLite storage plugin. This setting is equivalent to running the following statements when opening a SQLite file: `PRAGMA journal_mode=WAL; PRAGMA synchronous=NORMAL`. This removes the risk of an entire `.db3` file being corrupted, at the cost of performance. By using a write-ahead log, only the messages in the last few transactions may be lost if the data from those transactions were not fully committed to disk. Therefore in this mode, SQLite3 offers similar guarantees against corruption as MCAP.
+- `mcap_default`: The default recording configuration of the MCAP storage plugin. Produces un-compressed, chunked MCAPs with a message index for efficient read performance.
+- `mcap_nochunking`: The highest-throughput configuration of the MCAP storage plugin. Does not write a message index to the MCAP, so bags recorded with this mode need to be reindexed later to be read efficiently.
+- `mcap_uncompressed_crc`: Like `mcap_default`, but calculates CRCs for each chunk so that readers can identify if a chunk contains corrupted data. This feature does not have an equivalent in the SQLite3 storage plugin.
+- `mcap_compressed_nocrc`: Like `mcap_default`, but compressed each chunk with `zstd` compression on default settings.
+
+## Software Versions
+
+Some installed package versions were omitted for clarity.
+
+| ROS 2 Package Name                | Version |
+| --------------------------------- | ------- |
+| mcap_vendor                       | 0.5.0   |
+| rcl                               | 5.4.0   |
+| rmw                               | 6.3.0   |
+| rmw_dds_common                    | 1.7.0   |
+| rmw_fastrtps_cpp                  | 6.3.0   |
+| rmw_fastrtps_shared_cpp           | 6.3.0   |
+| rmw_implementation                | 2.9.0   |
+| rmw_implementation_cmake          | 6.3.0   |
+| ros_core                          | 0.10.0  |
+| ros_environment                   | 3.2.0   |
+| ros_workspace                     | 1.0.2   |
+| rosbag2_compression               | 0.17.0  |
+| rosbag2_cpp                       | 0.17.0  |
+| rosbag2_storage                   | 0.17.0  |
+| rosbag2_storage_default_plugins   | 0.17.0  |
+| rosbag2_storage_mcap              | 0.5.0   |
+| rosbag2_storage_plugin_comparison | 0.1.0   |
+| rosbag2_storage_sqlite3           | 0.17.0  |
+| rosbag2_test_common               | 0.17.0  |
+| shared_queues_vendor              | 0.17.0  |
+| sqlite3_vendor                    | 0.17.0  |
+| zstd_vendor                       | 0.17.0  |
+
+## Message sizes tested
+
+- 1MiB
+- 10KiB
+- 100B
+- Mixed: Messages from all of the above sizes at the following ratios (by number of bytes, not messages)
+  - 1MiB: 70%
+  - 10KiB: 20%
+  - 100B: 10%
```

Comment:
Here would be great to add some notes about using abbreviations at least for `MiB`.
For me it wasn't clear that MiB is a `Mebibyte` not a `Megabit` and it is equal to 2^20  or 1,048,576 bytes. A megabyte is equal to 10^6 1,000,000 bytes.